muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/communicator/CommunicatorAutoReply.java

```diff
@@ -0,0 +1,112 @@
+              DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.ENGLISH);
+              start = LocalDate.parse(startProperty.getValue(), formatter);
+              end = LocalDate.parse(endProperty.getValue(), formatter);
+            }
+            if (start != null && end != null) {  
+              LocalDate now = LocalDate.now();
+              Date notificationDate = new Date();
+              if (start.equals(now) || (now.isAfter(start) && now.isBefore(end)) || end.equals(now)) {
+                
+                DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("dd.MM.yyy", Locale.ENGLISH);
+                String startDate = outputFormatter.format(start);
+                String endDate = outputFormatter.format(end);
+                
+                if (autoReplyMsg != null) {
+                  replyMessage = autoReplyMsg.getValue();
+                } else {
+                  replyMessage = "Automaattinen vastaus: Poissa " + startDate + " - " + endDate;
+                }
+                if (autoReplySubject != null) {
+                  replySubject = autoReplySubject.getValue();
+                } else {
+                  replySubject = "Automaattinen vastaus: Poissa " + startDate + " - " + endDate;
+                }
+                List<UserEntity> recipientsList = new ArrayList<UserEntity>();
+                recipientsList.add(sender);
+                VacationNotifications vacationNotification = communicatorController.findVacationNotification(sender, recipient);
+                
+                if (vacationNotification != null) {
+                  Date lastNotification = vacationNotification.getNotificationDate();
+
+                  LocalDate lastPlusFour = lastNotification.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+
+                  lastPlusFour = lastPlusFour.plusDays(4);
+                  Date fourDays = java.sql.Date.valueOf(lastPlusFour);
+
+                  if (java.sql.Date.valueOf(now).after(fourDays)) {
+                    communicatorController.createVacationNotification(sender, recipient, notificationDate);
+                    communicatorController.replyToMessage(recipient, communicatorMessage.getCategory().getName(), replySubject, replyMessage, recipientsList, communicatorMessage.getCommunicatorMessageId());
+                  }
+                } else {
+                  communicatorController.createVacationNotification(sender, recipient, notificationDate);
+                  communicatorController.replyToMessage(recipient, communicatorMessage.getCategory().getName(), replySubject, replyMessage, recipientsList, communicatorMessage.getCommunicatorMessageId());
+                }
+              }
+            }
+          }
+        }
+      }
+    } else {
+      logger.log(Level.SEVERE, String.format("Communicator couldn't send automatic reply as some entity was not found"));
```

Comment:
Just as a side notice, logging the lack of each piece of required information separately (earlier remark) makes it easier to debug problems since we don't have to guess which entity was missing.